pandas/core/internals/construction.py

```diff
@@ -304,10 +307,33 @@ def ndarray_to_mgr(
     index, columns = _get_axes(
         values.shape[0], values.shape[1], index=index, columns=columns
     )
-    values = values.T
 
     _check_values_indices_shape_match(values, index, columns)
 
+    if typ == "array":
+
+        values = sanitize_to_nanoseconds(values)
+        if issubclass(values.dtype.type, str):
+            values = np.array(values, dtype=object)
+
+        if dtype is None and is_object_dtype(values.dtype):
+            arrays = [
+                ensure_wrapped_if_datetimelike(
+                    maybe_infer_to_datetimelike(values[:, i].copy())
+                )
+                for i in range(values.shape[1])
+            ]
+        elif is_datetime_or_timedelta_dtype(values.dtype):
+            arrays = [
+                ensure_wrapped_if_datetimelike(values[:, i].copy())
+                for i in range(values.shape[1])
+            ]
+        else:
+            arrays = [values[:, i].copy() for i in range(values.shape[1])]
+        return ArrayManager(arrays, [index, columns], verify_integrity=False)
```

Comment:
could do the ensure_wrapped_if_datetimelike just once down here.  i think thatd make the sanitize_to_nanoseconds unnecessary too

Reply:
> could do the ensure_wrapped_if_datetimelike just once down here. 

I would like to avoid calling it on each column if the original dtype of the ndarray is not datetimelike (in which case it's never needed), so that's why I did it with an `elif is_datetime_or_timedelta_dtype(values.dtype`) check above.

But indeed, because I already call `ensure_wrapped_if_datetimelike` now, the `sanitize_to_nanoseconds` should no longer be necessary, I think.

Reply:
fair enough.  i think in the `elif is_datetime_or_timedelta_dtype(values.dtype)` block you could call ensure_wrapped_if_datetimelike only once on `values` and then slice.

Reply:
Yep, good idea, updated.